renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -2021,6 +1986,13 @@ bool WrappedVulkan::Serialise_vkCmdNextSubpass(SerialiserType &ser, VkCommandBuf
                                              FindCommandQueueFamily(m_LastCmdBufferID),
                                              (uint32_t)imgBarriers.size(), imgBarriers.data());
       }
+      else if(IsRenderpassOpen(m_LastCmdBufferID) && m_FirstEventID != m_LastEventID)
+      {
+        commandBuffer = RerecordCmdBuf(m_LastCmdBufferID);
+        ObjDisp(commandBuffer)->CmdNextSubpass(Unwrap(commandBuffer), contents);
+
+        m_BakedCmdBufferInfo[m_LastCmdBufferID].endBarriers.append(GetImplicitRenderPassBarriers());
```

Comment:
I _think_ we also need a `GetCmdRenderState().subpass++;` in this path right? `GetImplicitRenderPassBarriers()` looks up the current subpass based on that.

Reply:
Good catch, however as discussed, 

`GetCmdRenderState().subpass++`

actually modifies the vulkan pipeline state of the selected drawcall in the case of a partial replay, so I moved the active subpass count to a new variable which now GetImplicitRenderPassBarriers reads.